app/src/main/kotlin/nyc/jsjrobotics/palmrgb/androidInterfaces/DefaultActivity.kt

```diff
@@ -11,36 +11,71 @@ import io.reactivex.disposables.CompositeDisposable
 import nyc.jsjrobotics.palmrgb.Application
 import nyc.jsjrobotics.palmrgb.R
 import nyc.jsjrobotics.palmrgb.customViews.SubActivityToolbar
+import nyc.jsjrobotics.palmrgb.firebase.SimpleAuth
+import nyc.jsjrobotics.palmrgb.firebase.eventlisteners.MessageEventListener
 import nyc.jsjrobotics.palmrgb.fragments.NavigationBarSettings
+import nyc.jsjrobotics.palmrgb.toast
 import javax.inject.Inject
+import kotlin.properties.Delegates
 
 /**
  * Default Activity subscribes to abtest changes for recreate and
- * manages connection state for BrokenHungryBackground
+ * manages connection state for PalmRgbBackground
  */
-abstract class DefaultActivity : FragmentActivity() , IDefaultActivity {
+abstract class DefaultActivity : FragmentActivity(), IDefaultActivity {
 
     private val disposables: CompositeDisposable = CompositeDisposable()
 
     @Inject
-    lateinit var navigationBarSettings : NavigationBarSettings
+    lateinit var navigationBarSettings: NavigationBarSettings
+
+    @Inject
+    lateinit var messageEventListener: MessageEventListener
+
+    @Inject
+    lateinit var simpleAuth: SimpleAuth
 
     override fun getActivity(): FragmentActivity = this
 
     override fun applicationContext(): Context = applicationContext
 
+    var activityInBackground : Boolean by Delegates.notNull()
```

Comment:
Remove this variable. We do not save state of the activity lifecycle, we instead respond to it. If the activity is in the background, you should unsubscribe the disposable. Hence, in onResume, not onCreate you should call `subscribeToIncomingMessages`. In onPause you should unsubscribe